pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -73,13 +91,19 @@ func (f *Function) validate() error {
 	if !IsNodeZero(&f.Config) {
 		config := yaml.NewRNode(&f.Config)
 		if _, err := config.GetMeta(); err != nil {
-			return fmt.Errorf("functionConfig must be a valid KRM resource with `apiVersion` and `kind` fields")
+			return &ValidateError{
+				Field:  fmt.Sprintf("pipeline.%s[%d].config", fnType, idx),
+				Reason: "functionConfig must be a valid KRM resource with `apiVersion` and `kind` fields",
+			}
 		}
 		configFields = append(configFields, "config")
 	}
 	if len(configFields) > 1 {
-		return fmt.Errorf("following fields are mutually exclusive: 'config', 'configMap', 'configPath'. Got %q",
-			strings.Join(configFields, ", "))
+		return &ValidateError{
+			Field: fmt.Sprintf("pipeline.%s[%d]", fnType, idx),
+			Reason: fmt.Sprintf("following fields are mutually exclusive: 'config', 'configMap', 'configPath'. Got %q",
```

Comment:
We can reword "mutually exclusive" to make it more human :)
"only one of `config`, `configMap`, `configPath` must be specified. Got %q"

Reply:
done